File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> hash = new HashSet<>(nums.length);
        for(int i : nums)
        {
            hash.add(i);
        }

        List<Integer> cands = new ArrayList<>();

        for(int i : nums)
        {
            if(!hash.contains(i-1))
                cands.add(i);
        }

        int res = 0;

        for(int i : cands)
        {
            int temp = i, len = 0;
            while(hash.contains(temp)){
                temp++;
                len++;
            }

            res = Math.max(len, res);
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        n = len(s) 
        if n == 1:
            return 1 

        res = 1
        charSet = set(s)

        for c in charSet:
            # sliding window 
            l = 0 
            temp = 0 
            for r in range(n):
                if s[r] != c:
                    temp += 1
                
                while temp > k:
                    if s[l] != c:
                        temp -= 1
                    
                    l += 1
                
                res = max(r - l + 1, res)

        return res 
